community/neo4j/src/test/java/org/neo4j/index/backup/IndexBackupIT.java

```diff
@@ -146,20 +145,25 @@ public void snapshotFilesDeletedWhenSnapshotReleased() throws IOException
     private void compareSnapshotFiles( Set<String> firstSnapshotFileNames, Set<String> secondSnapshotFileNames,
             FileSystemAbstraction fileSystem )
     {
-        assertThat(
-                format( "Should have at least %d modified index files. Snapshot files  are: %s", NUMBER_OF_INDEXES + 1,
-                        firstSnapshotFileNames ), firstSnapshotFileNames,
-                hasSize( greaterThanOrEqualTo( NUMBER_OF_INDEXES + 1 ) ) );
-        for ( String fileName : firstSnapshotFileNames )
+        Set<String> firstSnapshotIndexesSegments = firstSnapshotFileNames.stream()
+                        .filter( this::segmentsFilePredicate )
```

Comment:
The filter might be moved to `getFileNames` instead. Not sure what other implications this has for the rest of the test though.